src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1930,7 +2174,17 @@ class MutateSomePartColumnsTask : public IExecutableTask
             }
         }
 
-        MutationHelpers::finalizeMutatedPart(ctx->source_part, ctx->new_data_part, ctx->execute_ttl_type, ctx->compression_codec, ctx->context, ctx->metadata_snapshot, ctx->need_sync);
+        for (const auto & mutation_context : ctx->projection_mutation_contexts)
+        {
+            if (mutation_context->kind != ProjectionMutationContext::MutationKind::SOME)
+                continue;
```

Comment:
Why? I think it should not be possible according to the code in ``, so it's better to throw logical error here.